common-lib/src/main/scala/com/gu/mediaservice/GridClient.scala

```diff
@@ -132,73 +121,84 @@ class GridClient(apiKey: String, services: Services, maxIdleConnections: Int, de
-      val downstreamErrorMessage = res.bodyAsString
-
-      val errorJson = Json.obj(
-        "level" -> "ERROR",
-        "errorStatusCode" -> statusCode,
-        "message" -> Json.obj(
-          "errorMessage" -> errorMessage,
-          "downstreamErrorMessage" -> downstreamErrorMessage
-        )
-      )
-      logger.error(errorJson.toString())
-      throw new DownstreamApiInBadStateException(errorMessage, downstreamErrorMessage)
+  private def validateResponse[T](
+                                   response: WSResponse,
+                                   url: URL,
+                                   foundFn: ResponseWrapper => Option[T],
+                                   notFoundFn: ResponseWrapper => Option[T],
+                                   errorFn: Option[ResponseWrapper => Exception]
+                                 ): Option[T] = {
+    val res = processResponse(response, url)
+    res.statusCode match {
+      case 200 => foundFn(res)
+      case 404 => notFoundFn(res)
+      case failCode => errorFn match {
+        case Some(fn) => throw fn(res)
+        case _ =>
+          val errorMessage = s"Downstream API Failure: $url is in a bad state, code: $failCode"
+          val downstreamErrorMessage = res.bodyAsString
+
+          val errorJson = Json.obj(
+            "level" -> "ERROR",
+            "errorStatusCode" -> failCode,
+            "message" -> Json.obj(
+              "errorMessage" -> errorMessage,
+              "downstreamErrorMessage" -> downstreamErrorMessage
+            )
+          )
+          logger.error(errorJson.toString())
+          throw new DownstreamApiInBadStateException(errorMessage, downstreamErrorMessage)
+      }
     }
   }
 
-  def getImageLoaderProjection(mediaId: String, imageLoaderEndpoint: String): Option[Image] = {
+  def getImageLoaderProjection(mediaId: String, imageLoaderEndpoint: String, authFn: WSRequest => WSRequest)
+                              (implicit ec: ExecutionContext): Future[Option[Image]] = {
     logger.info("attempt to get image projection from image-loader")
-    val url = new URL(s"${imageLoaderEndpoint}/images/project/$mediaId")
-    val res = makeGetRequestSync(url, apiKey)
-    validateResponse(res, url)
-    logger.info(s"got image projection from image-loader for $mediaId with status code $res.statusCode")
-    if (res.statusCode == 200) Some(res.body.as[Image]) else None
+    val url = new URL(s"$imageLoaderEndpoint/images/project/$mediaId")
+    makeGetRequestAsync(
+      url,
+      authFn,
+      {res:ResponseWrapper => Some(res.body.as[Image])},
+      {_:ResponseWrapper => None}
+    )
   }
 
-  def getLeases(mediaId: String)(implicit ec: ExecutionContext): Future[LeasesByMedia] = {
+  def getLeases(mediaId: String, authFn: WSRequest => WSRequest)(implicit ec: ExecutionContext): Future[Option[LeasesByMedia]] = {
```

Comment:
For all of these endpoints I think that it is possible to eliminate the `Option` from the return type as None is never used to signify anything useful. Future can be `Failure` if necessary, and probably should be as if there is an upstream failure. 

I've not yet found a reason this needs to be option. I think the superflous option stems from the functions that are passed into the request function (discussed above) - the suggested refactor will eliminate this but even without that you can remove Option from all of the functions.

Reply:
Done.   Took a little thinking :)